### ISSUE TITLE
NodeNotFoundError while building from list

### ISSUE THREAD

**Issue by asarkar** [joowani/binarytree#31]
`[1, 2, 3, None, None, 5, 6, 7, 8, 9, None, None, None, 10, None, 11]` should produce:
<img width="362" alt="Screen Shot 2021-03-26 at 3 58 11 AM" src="https://user-images.githubusercontent.com/1302775/112621952-957ea980-8de7-11eb-9e43-f81ea941520c.png">

`[2, 5, None, 3, None, 1, 4]` should produce:
<img width="145" alt="Screen Shot 2021-03-26 at 3 58 28 AM" src="https://user-images.githubusercontent.com/1302775/112622034-b1824b00-8de7-11eb-971e-2bcf71cd9215.png">

but both fail.
```
                if parent is None:
>                   raise NodeNotFoundError(
                        "parent node missing at index {}".format(parent_index)
                    )
E                   binarytree.exceptions.NodeNotFoundError: parent node missing at index 3

/Users/me/.pyenv/versions/3.9.2/lib/python3.9/site-packages/binarytree/__init__.py:2008: NodeNotFoundError
```

Whereas the following code builds the trees as expected:
```
def from_list(nums: Sequence[Optional[int]]) -> Optional[TreeNode]:
    """Build a binary tree in level order.
    The idea is to take two numbers and associate with the first non-null parent that hasn't been looked at.

    Keyword arguments:
    nums -- node values
    """
    queue: collections.deque[TreeNode] = collections.deque()
    root = None
    if nums:
        root = TreeNode(nums[0])
        queue.append(root)
    i = 1

    while i < len(nums):
        node = queue.popleft()
        if nums[i] is not None:
            node.left = TreeNode(nums[i])
            queue.append(node.left)
        i += 1
        if i < len(nums) and nums[i] is not None:
            node.right = TreeNode(nums[i])
            queue.append(node.right)
        i += 1

    queue.clear()
    return root
```

**Comment by joowani** [comment]
Hi @asarkar,

The `binarytree.build` function uses a different representation explained [here](https://en.wikipedia.org/wiki/Binary_tree#Arrays) (also in the function docstring and binarytree [docs](https://binarytree.readthedocs.io/en/main/specs.html#binarytree.build)).

I've also seen the method you're suggesting. It does seem to have its own usecases and benefits (e.g. more compact form) so I can add it to the library as `binarytree.build2`. Thanks!

**Comment by asarkar** [comment]
The tree representation I shown is widely used online, including LeetCode, although they don’t disclose the implementation. I’m happy to make a PR if it’s not too much trouble.

**Comment by joowani** [comment]
Thank you for the offer! But I just released [6.3.0](https://github.com/joowani/binarytree/releases/tag/6.3.0) with new methods to support the list representation you suggested. Please feel free to try it out.

And yes the original approach will use more space but the point of it is its indexing properties. It allows efficient lookup and storage using arrays (no need for higher level structs). Heaps often use this form as mentioned in the wiki page.
